test/gtest/ucp/test_ucp_am.cc

```diff
@@ -1291,6 +1295,24 @@ class test_ucp_am_nbx_rndv : public test_ucp_am_nbx {
         return UCS_INPROGRESS;
     }
 
+    static ucs_status_t
+    am_data_deferred_rndv_cb(void *arg, const void *header,
```

Comment:
looks like there is already the callback you need - `am_data_deferred_reject_rndv_cb`
Please check `deferred_reject_rndv` test. You can do the same, but send 0-bytes and the receive it

Reply:
good catch, done